### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by davetcoleman**
See commit descriptions for the change log (I committed everything separately in small increments). This is mostly a cleanup commit, but includes removing old functions that are no longer used and replacing some pointers with boost shared_ptrs

### CODE REVIEW

File: gazebo_ros/include/gazebo/gazebo_ros_api_plugin.h [hunk 1]
```diff
@@ -231,19 +234,79 @@ class GazeboRosApiPlugin : public SystemPlugin
...
+  /// \brief
+  bool spawnAndConform(TiXmlDocument &gazebo_model_xml, std::string model_name, gazebo_msgs::SpawnModel::Response &res);
+
   /// \brief helper function for applyBodyWrench
+  ///        shift wrench from reference frame to target frame
   void transformWrench(gazebo::math::Vector3 &target_force, gazebo::math::Vector3 &target_torque,
                        gazebo::math::Vector3 reference_force, gazebo::math::Vector3 reference_torque,
                        gazebo::math::Pose target_to_reference );
+
+  /// \brief Used for the dynamnic reconfigure callback function template
```
Comment (ffurrer) [inline gazebo_ros/include/gazebo/gazebo_ros_api_plugin.h]: typo dynamnic

File: gazebo_ros/include/gazebo/gazebo_ros_api_plugin.h [hunk 2]
```diff
@@ -231,19 +234,79 @@ class GazeboRosApiPlugin : public SystemPlugin
...
+  void physicsReconfigureCallback(gazebo::PhysicsConfig &config, uint32_t level);
+
+  /// \brief waits for the rest of Gazebo to be ready before initializing the dynamic reconfigure services
+  void physicsReconfigureThread();
+
+  /// \brief Unusd
```
Comment (ffurrer) [inline gazebo_ros/include/gazebo/gazebo_ros_api_plugin.h]: another typo... i hope i don't end up only correcting typos ;)

File: gazebo_plugins/src/gazebo_ros_joint_trajectory.cpp [hunk 3]
```diff
@@ -222,8 +222,8 @@ void GazeboRosJointTrajectory::FixLink(physics::LinkPtr link)
     this->joint_ = this->world_->GetPhysicsEngine()->CreateJoint("revolute",this->model_);
     this->joint_->SetModel(this->model_);
     math::Pose pose = link->GetWorldPose();
-    //math::Pose  pose(math::Vector3(0, 0, 0.2), math::Quaternion(1, 0, 0, 0));
-    this->joint_->Load(physics::LinkPtr(), link, pose.pos);
+    //this->joint_->Load(physics::LinkPtr(), link, pose.pos);
```
Comment (ffurrer) [inline gazebo_plugins/src/gazebo_ros_joint_trajectory.cpp]: why only comment this and not delete it?

File: gazebo_ros/include/gazebo/gazebo_ros_api_plugin.h [hunk 4]
```diff
@@ -231,19 +234,79 @@ class GazeboRosApiPlugin : public SystemPlugin
...
+  void onResponse(ConstResponsePtr &response);
+
+  /// \brief utilites for checking incoming string URDF/SDF/Param
+  bool isURDF(std::string model_xml);
+
+  /// \brief utilites for checking incoming string URDF/SDF/Param
+  bool isSDF(std::string model_xml);
+
+  /// \brief utilites for checking incoming string URDF/SDF/Param
```
Comment (ffurrer) [inline gazebo_ros/include/gazebo/gazebo_ros_api_plugin.h]: and three times utilities instead of utilites

**Comment by ffurrer** [comment]
looks good to me

File: gazebo_plugins/src/gazebo_ros_joint_trajectory.cpp (hunk 3 above)
Comment (davetcoleman) [inline gazebo_plugins/src/gazebo_ros_joint_trajectory.cpp]: I wasn't sure if it was the correct fix, then I forgot to remove it after compile. Fixed & thanks!

**Comment by davetcoleman** [comment]
Thanks, fixes per ffurrer:
https://github.com/osrf/gazebo_ros_pkgs/commit/983e165a82e19f0c3b5d0ba0f7773c2593841c15
https://github.com/osrf/gazebo_ros_pkgs/commit/ffdc0d5261d5ba4aeb2771fb33c3edd11307bcac

**Pull request merged**